chalicelib/checks/wrangler_checks.py

```diff
@@ -1099,3 +1099,103 @@ def share_core_project(connection, **kwargs):
+    action search through variant samples with genes belonging to all current
+    gene lists and add them to the indexing queue if the gene lists are not
+    embedded.
+    """
+
+    check = CheckResult(connection, 'update_variant_genelist')
+    genelist_search = ff_utils.search_metadata(
+        'search/?type=GeneList&field=uuid&field=genes.uuid', 
+        key=connection.ff_keys
+    )
+    variant_samples_to_index = []
+    for genelist in genelist_search:
+        batch = []
+        for idx in range(len(genelist['genes'])):
+            batch.append(genelist['genes'][idx]['uuid'])
+            if len(batch) == 40 or idx == (len(genelist['genes']) - 1):
+                batch_terms = [
+                    '&variant.genes.genes_most_severe_gene.uuid=' + uuid
+                    + '&variant.genes.genes_most_severe_gene.gene_lists.uuid!='
+                    + genelist['uuid']
+                    for uuid in batch
+                ]
+                variant_sample_search_term = (
+                    'search/?type=VariantSample' + ''.join(batch_terms) 
+                    + '&field=uuid'
+                )
+                variant_sample_search = ff_utils.search_metadata(
+                    variant_sample_search_term,
+                    key=connection.ff_keys
+                )
+                variant_samples_to_index += [
+                    variant_sample['uuid'] for variant_sample in
+                    variant_sample_search
+                ]
+                batch = []
+    items_to_index = list(set(variant_samples_to_index))
+    if items_to_index:
+        check.status = 'WARN'
+        check.summary = (
+                'Some variants/variant samples need to be re-indexed to '
+                'update gene lists.'
+        )
+        check.description = (
+                '{} variants/variant samples need to be re-indexed to update '
+                'gene lists.'.format(len(items_to_index))
+        )
+        check.full_output = items_to_index
+        check.allow_action = True
+        check.action = 'queue_variants_to_update_genelist'
+    else:
+        check.status = 'PASS'
+        check.summary = (
+                'All variants and variant samples have up-to-date gene lists.'
+        )
+        check.description = check.summary
+    return check
+
+@action_function()
+def queue_variants_to_update_genelist(connection, **kwargs):
+    """
+    Add variant samples to indexing queue to update gene lists. 
+
+    Works with output of update_variant_genelist() above. 
+    """
+
+    action = ActionResult(connection, 'update_variant_genelist')
+    check_response = action.get_associated_check_result(kwargs)
+    check_full_output = check_response['full_output']
+    queue_index_post = {
+        'uuids': check_full_output,
+        'target_queue': 'primary',
+        'strict': True
+    }
+    post_url = 'https://cgap.hms.harvard.edu/queue_indexing'
```

Comment:
instead of using the full post url you might want to get the server from the connection arg and append `queue_indexing` to that, so theoretically we could do this on a test environment also.

Reply:
Done